packages/angular/tsconfig.ngc.json

```diff
@@ -0,0 +1,28 @@
+/* To learn more about this file see: https://angular.io/config/tsconfig. */
+// This tsconfig is used when building @sentry/angular with the Angular
+// compiler and `ng-packagr`. It configures a production build conforming
+// to the Angular Package Format (APF).
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "target": "es2015",
+    "lib": [
+      "dom",
+      "es2018"
+    ],
+    "baseUrl": "./",
+  },
+  "angularCompilerOptions": {
+    "skipTemplateCodegen": true,
+    "strictMetadataEmit": true,
+    "enableResourceInlining": true,
```

Comment:
Have you considered more strict settings?
https://angular.io/guide/angular-compiler-options

Reply:
Up to this point, I went with the default parameters included by the Angular CLI when setting up an Angular library via the CLI (with Angular 10). Do you have suggestions what we should add here (keeping in mind Angular 10-13 support)?

Reply:
Have you run this setup with `ng new` or `ng new --strict`?

Specifically, the strict flag does the following:

- Enables strict mode in TypeScript
- Turns template type checking to Strict
- Default bundle budgets have been reduced by ~75%
- Configures linting rules to [prevent declarations of type ](https://palantir.github.io/tslint/rules/no-any/)[any](https://palantir.github.io/tslint/rules/no-any/)
- Configures your app as side-effect free to enable more advanced tree-shaking

Reply:
I see, indeed I set it up just with `ng new`. I'll look into this and I'll see what we can do on this end. 

Just to be sure though, using the `--strict` flag would just add additional parameters to the tsconfig? And since Angular seems to make a few differences in the tsconfig when setting up libraries instead of apps, are the stricter settings also recommended for library creation? Asking because I find it odd that the Angular library creation guide does not mention them.

Reply:
Thanks for the links and for the expanations. Some of the TS specific strict settings I see in the linked diff `tsconfig.json` are already part of our general `tsconfig` which also applies to our Angular SDK (see https://github.com/getsentry/sentry-javascript/blob/master/packages/typescript/tsconfig.json). So that's at least something. 

For the remaining ones and Angular specific ones, I'll try to add them and check if they break something on our end. Can't tell you yet if it will happen in this PR or a subsequent one. If you see something missing once we release the new Angular SDK with v7, please feel free to open up an issue. Of course, a PR would be welcome as well (always and not only limited to this aspect of our SDKs) ;)

Reply:
While checking out the `--strict` flag in `ng-new`, I found that certain options are only applied to angular applications (`ng g application`) and not to libraries (`ng g library`). E.g. budgets.

Some others, such as `strictTemplates: true` apply to both. Gonna include them.